### DISCUSSION TITLE
Do not add new window to group, if group is the only active window

### DISCUSSION THREAD

**Discussion by JoseConseco** [hyprwm/Hyprland#938] [General]
Problem:
- I have one full screen group active (it is the only visible window)
- when opening new app - it is automatically added to group -  but I want it to be tiled next to group
- now I have to ungroup, pick old group windows, group, switch back to new window  ;( 
Is there way to prevent auto adding of new opened app to group - if it is the only active window on monitor? 
I can see all works as expected, if we have two windows  [grouped]**[single-active]**, then opening new app wont group it obviously (group was not active).

**Comment by vaxerski** [comment]
No way to do that atm, unless you write a script. Then it might work.

**Reply by JoseConseco** [reply]
Ok.  It would be great if in future hyprland had option to exclude new window from auto-adding to group. For me, in most cases when I have setup like:
**[A B]** [C] ,    -- [A B] group is active, 
Then I wish adding D would give 
[A B]**[D]** [C]         -- auto split window tree with [AB] and [D] as new leafs - if possible
Rather than
[A B **D**][C]  
Hope you will consider adding toggle  for this.

**Comment by JoseConseco** [comment]
ok, I kind of got it working with:

```bash
#!/bin/sh

function handle {
  # echo "Handling $1"
  if [[ ${1:0:10} == "openwindow" ]]; then
    echo "Opening window"
    echo "Handling $1"
    hyprctl dispatch togglegroup none
    hyprctl dispatch cyclenext none
    # sleep 0.1
    hyprctl dispatch togglegroup none
    hyprctl dispatch cyclenext none
  fi
  if [[ ${1:0:12} == "activewindow" ]]; then
    echo "active window"
    echo "Handling $1"
    # hyprctl dispatch cyclenext
  fi
}

socat - UNIX-CONNECT:/tmp/hypr/$(echo $HYPRLAND_INSTANCE_SIGNATURE)/.socket2.sock | while read line; do handle $line; done

```
How to get info, - if previous window is group (before new window is added) ?  I could disable then above code for non groups.
